server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java

```diff
@@ -307,10 +307,11 @@ private Path cleanServerDirectory(Path serverOutputPath) {
                   cmd.getHostConfig().withMemorySwap(IMAGE_MEMORY_SWAP);
                }
             });
-
-      // Replace 99 with index of server to debug
-      if (i == 99) {
-         container.withEnv("JAVA_OPTS", debugJvmOption());
+      String debug = configuration.properties().getProperty(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_DEBUG);
+      if (debug != null && Integer.parseInt(debug) == i) {
```

Comment:
catch `NumberFormatException`?

Reply:
No need: it will break the test, which is good enough as a solution :)